### PR TITLE
remove call native `unzip` on non-windows

### PR DESCRIPTION
**Pull request by XaveScor**
If user hasn't `unzip` util on Linux or MacOS then flow-typed run-tests failed. I replace native call to unzip module.
I filter files in bin dir, if we have trash in bin folder.
#1519 should be solved

### CODE REVIEW

**Comment by XaveScor** [comment]
@GAntoine here?)

**Comment by GAntoine** [comment]
This one I'll need a little time to comb over, but I don't expect there to be any issues.

File: cli/src/commands/runTests.js [hunk 1]
```diff
@@ -260,15 +240,24 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
...
  * Return the sorted list of cached flow binaries that have previously been retrieved from github
  * and cached in the `.flow-bins-cache` directory.  This function is usually called when a failure
  * has occurred when attempting to refresh the flow releases from github, i.e. offline or over
  * API limit.
  */
 async function getCachedFlowBinVersions(): Promise<Array<string>> {
-  // read the files from the bin dir and remove the leading `flow-v` prefix
-  const versions = (await fs.readdir(path.join(BIN_DIR))).map(dir =>
+  // read the files with name `flow-vx.x.x` from the bin dir and remove the leading `flow-v` prefix
+  const versions = (await fs.readdir(path.join(BIN_DIR))).filter(checkFlowFilename).map(dir =>
```
Comment (GAntoine) [inline cli/src/commands/runTests.js]: What kind of trash are you trying to filter out? As in, which files are causing issues?

File: cli/src/commands/runTests.js [hunk 2]
```diff
@@ -99,10 +99,6 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
       const QUERY_PAGE_SIZE = 100;
       const OS_ARCH_FILTER_RE = new RegExp(BIN_PLATFORM);
 
-      if (!await fs.exists(BIN_DIR)) {
```
Comment (GAntoine) [inline cli/src/commands/runTests.js]: ~What's the reason for removing this check?~ Nevermind, I see you moved it below.

File: cli/src/commands/runTests.js [hunk 3]
```diff
@@ -208,38 +204,22 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
...
-            await new Promise((res, rej) => {
-              const unzipExtractor = unzip.Extract({ path: flowBinDirPath });
-              unzipExtractor.on("error", function(err) {
-                rej(err);
-              });
-              unzipExtractor.on("close", function() {
-                res();
-              });
-              fs.createReadStream(zipPath).pipe(unzipExtractor);
+          await new Promise((res, rej) => {
```
Comment (GAntoine) [inline cli/src/commands/runTests.js]: This change makes sense 👍

**Comment by GAntoine** [comment]
@XaveScor Sorry for taking so long to get to this, did a review and I have a couple questions.

File: cli/src/commands/runTests.js (hunk 1 above)
Comment (XaveScor) [reply cli/src/commands/runTests.js]: If `flow-typed run-tests` crushed BIN_DIR can contain 'garbage files'(such as flow-0.x.x.zip, etc). Without filter `runTests.js` tries check semver and failed. Semver returns undefined.

**Pull request merged**